### ISSUE TITLE
Enhancement: [member-ordering] allow regex pattern as a member type when initializing a member

### ISSUE THREAD

**Issue by mostapko** [typescript-eslint/typescript-eslint#8035]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/member-ordering/

### Description

Add an option to pass a regex pattern of member initialization value to the memberTypes array. It would be helpful for anyone who wants to customize the ordering of members while initializing a member and the default member types are not sufficient/descriptive enough.

One particular use case would be in newer versions of Angular where services are injected on top of the file with "private readonly"  declaration. Currently this results in an Eslint error because private declarations should be declared after public ones.

I am sure that there are other use cases where such option would be useful. 

### Fail

```TypeScript
Eslint Config fragment:
'@typescript-eslint/member-ordering': [
          'error',
          {
            default: {
              memberTypes: // pass the default memberTypes array
              }
           }
]


Typescript Code:
private readonly myService = inject(MyService);
public myVariable = 'Test';

// Results in: ESLint: Member myVariable should be declared before all private instance field definitions.(@typescript-eslint/member-ordering)
```

### Pass

```TypeScript
Eslint Config fragment:
'@typescript-eslint/member-ordering': [
          'error',
          {
            default: {
              memberTypes: [ /^inject/, ...defaultMemberTypesArray]
              }
           }
]

Typescript Code:
private readonly myService = inject(MyService);
public myVariable = 'Test';

// No errors because we added a regex for "inject"
```
